### PR TITLE
Add MIOpenDriver Xdlop tests

### PR DESCRIPTION
**Pull request by yiqian1**

### CODE REVIEW

**Review by jerryyin — APPROVED** [review]
~~Changes LGTM. If all CI pass + this `miopen-test` CI pass, please merge yourself.~~

File: mlir/utils/jenkins/miopen-tests/resnet50_fp16_nhwc_xdlop.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  rm -rf ~/.config/miopen/
+}
+
+export MIOPEN_FIND_MODE=1
+export MIOPEN_DEBUG_CONV_GEMM=0
+export MIOPEN_DRIVER_USE_GPU_REFERENCE=1
+cleanup
+
+# Fwd
+export MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmFwdXdlop
```
Comment (jerryyin) [inline mlir/utils/jenkins/miopen-tests/resnet50_fp16_nhwc_xdlop.sh]: Xdlops

File: mlir/utils/jenkins/miopen-tests/resnet50_fp16_nhwc_xdlop.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+bin/MIOpenDriver convfp16 -F 1 -n 256 -c 512 -H 28 -W 28 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver convfp16 -F 1 -n 256 -c 512 -H 7 -W 7 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver convfp16 -F 1 -n 256 -c 512 -H 7 -W 7 -k 512 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver convfp16 -F 1 -n 256 -c 64 -H 56 -W 56 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver convfp16 -F 1 -n 256 -c 64 -H 56 -W 56 -k 64 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver convfp16 -F 1 -n 256 -c 64 -H 56 -W 56 -k 64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+# FIXME bin/MIOpenDriver convfp16 -F 1 -n 256 -c 3 -H 230 -W 230 -k 64 -y 7 -x 7 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+
+## Wrw
+export MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmWrWXdlop
```
Comment (jerryyin) [inline mlir/utils/jenkins/miopen-tests/resnet50_fp16_nhwc_xdlop.sh]: Xdlops

File: mlir/utils/jenkins/miopen-tests/resnet50_fp32_nchw_xdlop.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  rm -rf ~/.config/miopen/
+}
+
+export MIOPEN_FIND_MODE=1
+export MIOPEN_DEBUG_CONV_GEMM=0
+export MIOPEN_DRIVER_USE_GPU_REFERENCE=1
+cleanup
+
+# Fwd
+export MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmFwdXdlop
```
Comment (jerryyin) [inline mlir/utils/jenkins/miopen-tests/resnet50_fp32_nchw_xdlop.sh]: Xdlops

File: mlir/utils/jenkins/miopen-tests/resnet50_fp32_nchw_xdlop.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+bin/MIOpenDriver conv -F 1 -n 256 -c 512 -H 7 -W 7 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
+bin/MIOpenDriver conv -F 1 -n 256 -c 512 -H 7 -W 7 -k 512 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
+bin/MIOpenDriver conv -F 1 -n 256 -c 64 -H 56 -W 56 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
+bin/MIOpenDriver conv -F 1 -n 256 -c 64 -H 56 -W 56 -k 64 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
+bin/MIOpenDriver conv -F 1 -n 256 -c 64 -H 56 -W 56 -k 64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
+# FIXME bin/MIOpenDriver conv -F 1 -n 256 -c 3 -H 230 -W 230 -k 64 -y 7 -x 7 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
+
+
+# Wrw
+export MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmWrWXdlop
```
Comment (jerryyin) [inline mlir/utils/jenkins/miopen-tests/resnet50_fp32_nchw_xdlop.sh]: Xdlops

File: mlir/utils/jenkins/miopen-tests/resnet50_fp32_nhwc_xdlop.sh [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  rm -rf ~/.config/miopen/
+}
+
+export MIOPEN_FIND_MODE=1
+export MIOPEN_DEBUG_CONV_GEMM=0
+export MIOPEN_DRIVER_USE_GPU_REFERENCE=1
+cleanup
+
+# Fwd
+export MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmFwdXdlop
```
Comment (jerryyin) [inline mlir/utils/jenkins/miopen-tests/resnet50_fp32_nhwc_xdlop.sh]: Xdlops

File: mlir/utils/jenkins/miopen-tests/resnet50_fp32_nhwc_xdlop.sh [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+bin/MIOpenDriver conv -F 1 -n 256 -c 512 -H 28 -W 28 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver conv -F 1 -n 256 -c 512 -H 7 -W 7 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver conv -F 1 -n 256 -c 512 -H 7 -W 7 -k 512 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver conv -F 1 -n 256 -c 64 -H 56 -W 56 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver conv -F 1 -n 256 -c 64 -H 56 -W 56 -k 64 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+bin/MIOpenDriver conv -F 1 -n 256 -c 64 -H 56 -W 56 -k 64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+# FIXME bin/MIOpenDriver conv -F 1 -n 256 -c 3 -H 230 -W 230 -k 64 -y 7 -x 7 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1 -I NHWC -O NHWC -f NHWC
+
+# Wrw
+export MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmWrWXdlop
```
Comment (jerryyin) [inline mlir/utils/jenkins/miopen-tests/resnet50_fp32_nhwc_xdlop.sh]: Xdlops

**Review by jerryyin — CHANGES_REQUESTED** [review]
Please investigate why CI pass with wrong solver name, [here](http://10.236.106.97:8080/job/MLIR-MIOpen-test/50/console).

**Comment by yiqian1** [comment]
The problem was I ran the shell scripts in the background and wait for all background jobs to finish. The background jobs continued even when one of tests failed and returned the status of the last executed process. I've changed the scripts to 

1) stop the script if one of background processes fails; 
2) count the number of successful tests and return exit code of 1 if one of tests fails.

**Review by jerryyin — APPROVED** [review]
LGTM. Thanks for helping triage && update the bash script such that test can work in parallel.  Please remember to make Jenkinsfile point back to `miopen-dialect` branch when this is merged.

**Pull request merged**